### PR TITLE
JOSS Paper Draft

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR contains the current draft of the JOSS paper. I figured this would be a good way of allowing people to provide their feedback on the paper, but you're also welcome to download the paper at https://github.com/kevin218/Eureka/actions/workflows/draft-pdf.yml (click on the most recent build, and then click "paper" under the Artifacts section to download the zip file) and send me feedback by email.

As a reminder, there is a ~1000 word limit which we are currently near (~900 words), so little text can be added without also removing text especially once we add in the figures and their captions.

### CODE REVIEW

File: joss/paper.md [hunk 1]
```diff
@@ -1,51 +1,120 @@
...
     orcid: 0000-0003-0514-1147
-    affiliation: X
+    affiliation: 8
 affiliations:
- - name: Max-Planck-Institut für Astronomie, Königstuhl 17, D-69117 Heidelberg, Germany
-   index: X
-date: xxx
+  - name: BAER Institute, NASA Ames Research Center, Moffet Field, CA 94035, USA
+    index: 1
+  - name: Centre for Exoplanets and Habitability, University of Warwick, Gibbet Hill Road, CV4 7AL Coventry, UK
```
Comment (evamariaa) [inline joss/paper.md:54]: Please change "Centre for Exoplanets and Habitability" to "Department of Physics", everything else is correct - thank you!

**Comment by lkreidberg** [comment]
hi Taylor, great work on this!!

I have a few very minor comments:

- line 23: what does "high quality" mean? I recommend a more specific description here ("precise, robust", etc)
- line 44-45: " although it is recommended that you skip the photom step in that pipeline." I think this is more detail than required here (this remark should stay in the Eureka! docs, but isn't a key feature of the code itself)
- line 60: "does more harm than good". This is a pretty strong statement! I recommend backing off here and just saying that generic pipelines are not sufficient for the extreme precision we require in our analyses
- line 64: "their" --> "his or her"
- line 67: "well documented" --> "well-documented"
- line 70: "analyses of HST observations" (no need to limit ourselves to WFC3)
- please add Ivelina Momcheva to the acknowledgments for useful discussions
- please also acknowledge Kreidberg et al. 2014 and the PACMAN pipeline (https://pacmandocs.readthedocs.io/en/latest/) for code used for Stage 5 fitting
- throughout: there are a few generic modifiers that could be dropped (line 30 "many", line 65 "quite")

File: docs/source/index.rst [hunk 2]
```diff
@@ -11,12 +11,12 @@ The code is not officially associated with JWST or the ERS team.
...
-- Stage 2: An optional step that calibrates Stage 1 data (performs flatfielding, unit conversion, etc.). This step can be skipped if you'd rather use STScI's JWST pipeline's Stage 2 outputs.
-- Stage 3: Starts with Stage 2 data and further calibrates (performs background subtraction, etc.) and reduces the data in order to convert 2D spectra into a time-series of 1D spectra.
-- Stage 4: Bins the 1D Spectra and generates light curves. Also removes 1D spectral drift/jitter and sigma clips outliers.
-- Stage 5: Fits the light curves with noise and astrophysical models.
-- Stage 6: Creates a table and a figure summarizing the transmission and/or emission spectra from your many fits.
+- Stage 1: An optional step that calibrates raw data (converts ramps to slopes for JWST observations). This step can be skipped if you'd rather use the ``jwst`` pipeline's Stage 1 outputs.
+- Stage 2: An optional step that further calibrates Stage 1 data (performs flat-fielding, unit conversion, etc. for JWST observations). This step can be skipped if you'd rather use the ``jwst`` pipeline's Stage 2 outputs, although it is recommended that you skip the photom step in that pipeline.
```
Comment (AarynnCarter) [inline docs/source/index.rst:15]: Perhaps:

"... Stage 2 outputs, although **at present** it is recommended..."

File: joss/paper.md [hunk 3]
```diff
@@ -1,51 +1,120 @@
...
...n and/or emission spectra. The pipeline has a modular structure with six stages, and each stage uses a "Eureka! Control File" (ECFs use the .ecf file extension) to allow for easy control of the pipeline's behavior. We have provided template ECFs for the MIRI, NIRCam, NIRISS, and NIRSpec instruments on JWST and the WFC3 instrument on the Hubble Space Telescope (HST). These templates give users a good starting point for their analyses, but `Eureka!` is not intended to be used as a black-box tool, and users should expect to fine-tune some settings for each observation in order to achieve optimal results. Throughout the pipeline, many intermediate figures and outputs are created to allow users to compare `Eureka!`'s performance using different parameter settings or to compare `Eureka!` with an independent pipeline. The ECF used to run each stage is also copied into the output folder from each stage to increase reproducibility. Finally, while `Eureka!` has been optimized for exoplanet observations (especially the later stages of the code), much of the core functionality could also be repurposed for JWST time-series observations in other research domains thanks to `Eureka!`'s modularity.
```
Comment (AarynnCarter) [inline joss/paper.md:81]: "uses a "Eureka! Control File" (ECFs use the .ecf file extension)" doesn't read quite right to me and the acronym isn't quite formally introduced. Maybe: "uses a "Eureka! Control File" (ECF; these all use the .ecf file extension)"

**Comment by AarynnCarter** [comment]
Great comments @lkreidberg! Although I just wanted to quickly mention that I think we should stick with "their" instead of "his or her" as it is more gender-inclusive.

File: docs/source/index.rst [hunk 4]
```diff
@@ -11,12 +11,12 @@ The code is not officially associated with JWST or the ERS team.
 
 The code is separated into six parts or "Stages":
 
-- Stage 1: An optional step that calibrates Raw data (converts ramps to slopes). This step can be skipped if you'd rather use STScI's JWST pipeline's Stage 1 outputs.
-- Stage 2: An optional step that calibrates Stage 1 data (performs flatfielding, unit conversion, etc.). This step can be skipped if you'd rather use STScI's JWST pipeline's Stage 2 outputs.
-- Stage 3: Starts with Stage 2 data and further calibrates (performs background subtraction, etc.) and reduces the data in order to convert 2D spectra into a time-series of 1D spectra.
-- Stage 4: Bins the 1D Spectra and generates light curves. Also removes 1D spectral drift/jitter and sigma clips outliers.
-- Stage 5: Fits the light curves with noise and astrophysical models.
-- Stage 6: Creates a table and a figure summarizing the transmission and/or emission spectra from your many fits.
+- Stage 1: An optional step that calibrates raw data (converts ramps to slopes for JWST observations). This step can be skipped if you'd rather use the ``jwst`` pipeline's Stage 1 outputs.
```
Comment (kevin218) [inline docs/source/index.rst:14]: Slightly cleaner version: "This step can be skipped within Eureka! if you would rather use the Stage 1 outputs from the ``jwst`` pipeline."

File: docs/source/index.rst [hunk 5]
```diff
@@ -11,12 +11,12 @@ The code is not officially associated with JWST or the ERS team.
...
+- Stage 2: An optional step that further calibrates Stage 1 data (performs flat-fielding, unit conversion, etc. for JWST observations). This step can be skipped if you'd rather use the ``jwst`` pipeline's Stage 2 outputs, although at present it is recommended that you skip the photom step in that pipeline.
```
Comment (kevin218) [inline docs/source/index.rst:15]: Make similar wording changes here for Stage 2

File: docs/source/index.rst [hunk 6]
```diff
@@ -11,12 +11,12 @@ The code is not officially associated with JWST or the ERS team.
...
+- Stage 3: Starts with the files produced by Stage 2 and performs background subtraction. This stage also reduces the data in order to convert 2D spectra into a time-series of 1D spectra or 2D photometric images into a 1D time-series.
```
Comment (kevin218) [inline docs/source/index.rst:16]: "Using Stage 2 outputs, performs background subtraction and optimal spectral extraction.  For spectroscopic observations, this stage generates a time series of 1D spectra.  For photometric observations, this stage generates a single light curve of flux versus time."

File: joss/paper.md [hunk 7]
```diff
@@ -1,51 +1,117 @@
...
+  - name: Caroline Piaulet
+    orcid: 0000-0002-2875-917X
+    affiliation: 9
+  - name: Joseph Filippazzo
+    orcid: 0000-0002-0201-8306
+    affiliation: 10
+  - name: Erin M. May
+    orcid: 0000-0002-2739-1465
+    affiliation: 11
+  - name: Kevin B. Stevenson
```
Comment (kevin218) [inline joss/paper.md:45]: If we're going by alphabetical order, I should be after Laura in the final group.

File: joss/paper.md [hunk 8]
```diff
@@ -1,51 +1,117 @@
...
...r emission spectra. The pipeline has a modular structure with six stages, and each stage uses a "Eureka! Control File" (ECF; these files use the .ecf file extension) to allow for easy control of the pipeline's behavior. We have provided template ECFs for the MIRI, NIRCam, NIRISS, and NIRSpec instruments on JWST and the WFC3 instrument on the Hubble Space Telescope (HST). These templates give users a good starting point for their analyses, but `Eureka!` is not intended to be used as a black-box tool, and users should expect to fine-tune some settings for each observation in order to achieve optimal results. Throughout the pipeline, intermediate figures and outputs are created to allow users to compare `Eureka!`'s performance using different parameter settings or to compare `Eureka!` with an independent pipeline. The ECF used to run each stage is also copied into the output folder from each stage to increase reproducibility. Finally, while `Eureka!` has been optimized for exoplanet observations (especially the later stages of the code), much of the core functionality could also be repurposed for JWST time-series observations in other research domains thanks to `Eureka!`'s modularity.
```
Comment (kevin218) [inline joss/paper.md:81]: Suggested changes (in bold) to improve flow and/or grammar:

- "with a particular focus on James Webb Space Telescope (JWST) **data**."
- "end-to-end pipeline **that** starts"
- "and ultimately **yields** precise exoplanet"
- "We **provide** template ECFs..."
- "**At each stage, the pipeline creates** intermediate figures and outputs **that** allow users..."
- "from each stage to **enhance** reproducibility"
- (especially the **latter** stages of the code)

Don't feel like you need to take all of these suggestions.

File: joss/paper.md [hunk 9]
```diff
@@ -1,51 +1,117 @@
...
+
 
+# Outline of `Eureka!`'s Stages
 
-# Statement of need
-Transit spectrsocopy is really hard!
+`Eureka!` is broken down into six stages which are as follows (also summarized in \autoref{fig:overview}):
 
+- Stage 1: An optional step that calibrates raw data (converts ramps to slopes for JWST observations). This step can be skipped if you'd rather use the `jwst` pipeline's Stage 1 outputs.
```
Comment (kevin218) [inline joss/paper.md:88]: Use the updated wording from `index.rst` for the Stage X descriptions

File: joss/paper.md [hunk 10]
```diff
@@ -1,51 +1,117 @@
...
+
 
+# Outline of `Eureka!`'s Stages
 
-# Statement of need
-Transit spectrsocopy is really hard!
+`Eureka!` is broken down into six stages which are as follows (also summarized in \autoref{fig:overview}):
```
Comment (kevin218) [inline joss/paper.md:86]: "into six stages, which are as follows"

File: joss/paper.md [hunk 11]
```diff
@@ -1,51 +1,117 @@
...
... pipeline from astronomical observatories is insufficiently precise for exoplanet time-series observations as the pipeline is more optimized for other science use cases. As such, it is common practice to develop a custom data analysis pipeline that starts from the original, uncalibrated images. Historically, data analysis pipelines have often been proprietary, so each new user of an instrument or telescope has had to develop their own pipeline. Also, clearly specifying the analysis procedure used can be challenging, especially with proprietary code, which erodes reproducibility. `Eureka!` seeks to be a next-generation data analysis pipeline for next-generation observations from JWST with open-source and well-documented code for easier adoption; modular code for easier customization while maintaining a consistent framework; and easy-to-use but powerful inputs and outputs for increased automation, increased reproducibility, and more thorough intercomparisons. By also allowing for analyses of HST observations within the same framework, it will also be possible to combine new and old observations to develop a more complete understanding of individual targets or even entire populations.
```
Comment (kevin218) [inline joss/paper.md:99]: - "**many of which will** significantly impact the final results."
- "is insufficiently **tailored** for exoplanet time-series observations..."
- "within the same framework, **users will be able** to combine new and old observations..."

**Review by kevin218 — COMMENTED** [review]
Paper looks to be in great shape!  Mostly minor comments relating to flow/grammar.  I think we should stick with using just the high-level flow chart in the paper and leave the detailed, stage-level flow charts for the documentation.  This is because there's a good chance that our detailed steps within each stage will change as we learn more about the instruments.

**Pull request closed without merging**